File: claude/enforce/tests/testdata/semgrep/cors-literal-wildcard_good.py
"""A credentialed FastAPI app pinned to one origin, and a public app with no credentials."""

from fastapi import FastAPI
from fastapi.middleware.cors import CORSMiddleware

APP_ORIGIN = "https://app.example.com"

app = FastAPI()
app.add_middleware(
    CORSMiddleware,
    allow_origins=[APP_ORIGIN],
    allow_credentials=True,
    allow_methods=["*"],
    allow_headers=["*"],
)

public_app = FastAPI()
public_app.add_middleware(
    CORSMiddleware,
    allow_origins=["*"],
    allow_credentials=False,
    allow_methods=["GET"],
)

File: claude/enforce/tests/testdata/semgrep/cors-literal-wildcard_good.ts
import cors from "cors";
import express from "express";

const APP_ORIGIN = "https://app.example.com";

const app = express();

app.use(cors({ origin: APP_ORIGIN, credentials: true }));

const publicApp = express();

publicApp.use(cors({ origin: "*", credentials: false }));

export { app, publicApp };
